packages/core/strapi/lib/commands/actions/develop/action.js

```diff
@@ -47,6 +47,9 @@ module.exports = async ({ build, watchAdmin, polling, browser }) => {
 };
 
 const primaryProcess = async ({ distDir, appDir, build, isTSProject, watchAdmin, browser }) => {
+  // Make sure generated types are deleted before building or re-starting the application
+  tsUtils.cleanupGeneratedTypes({ pwd: appDir });
```

Comment:
Not sure if we should cleanup on reload only for TS project or for both JS & TS. Open to suggestions

Same applies for [L.89-90](https://github.com/strapi/strapi/pull/17349/files#diff-e98af4f85f1bf3c732c5f74eb70bd7805dd439c6e459b1ea7e9cde63d2c4ec5dR89-R90)